app/connectors/AddressLookupConnector.scala

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import config.FrontendAppConfig
+import connectors.AddressLookupConnector._
+import models._
+import play.api.http.Status._
+import play.api.i18n.{Lang, Messages, MessagesApi}
+import play.api.libs.json.{__, Format, JsError, JsObject, JsSuccess, Json, Reads}
+import play.api.mvc.RequestHeader
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpReads, HttpResponse}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+import uk.gov.hmrc.http.HttpReads.Implicits._
+
+import scala.util.{Failure, Success, Try}
+
+class AddressLookupConnector @Inject()(http: HttpClient, appConfig: FrontendAppConfig)(
+  implicit ec: ExecutionContext
+) {
+
+  private val initUrl = appConfig.addressLookupFrontend.baseUrl + "/api/v2/init"
+  private val confirmedAddressUrl = appConfig.addressLookupFrontend.baseUrl + "/api/v2/confirmed"
+
+  def init(continueUrl: String, isUkAddress: Boolean)(implicit hc: HeaderCarrier, messages: Messages): Future[String] =
+    http
+      .POST[JsObject, HttpResponse](
+        initUrl,
+        initConfig(continueUrl, isUkAddress),
+        headers = List("Content-Type" -> "application/json")
+      )
+      .flatMap { response =>
+        Future.fromTry(
+          response.headers
+            .get("location")
+            .flatMap(_.headOption)
+            .fold[Try[String]](Failure(new RuntimeException("location header expected")))(Success(_))
```

Comment:
When running locally with ADDRESS_LOOKUP_FRONTEND v2.227.0 I keep getting this exception.